### PR TITLE
refactor: remove events support

### PR DESCRIPTION
**Pull request by mickhansen**
Will obviously not pass till the entire test suite has been refactored.

### CODE REVIEW

File: lib/promise.js [hunk 1]
```diff
@@ -57,236 +46,13 @@ Promise.prototype._then = function (didFulfill, didReject, didProgress, receiver
...
-    }
-    this.proxySql(ret);
-  }
-
-  return ret;
+  return bluebirdThen.call(this, didFulfill, didReject, didProgress, receiver, internalData);
 };
 
 var bluebirdSettle = Promise.prototype._settlePromiseAt;
 Promise.prototype._settlePromiseAt = function (index) {
```
Comment (mickhansen) [inline lib/promise.js]: @janmeier this can probably be removed too right?

Comment (janmeier) [inline lib/promise.js]: Yep

File: lib/promise.js [hunk 2]
```diff
@@ -34,7 +24,6 @@ var SequelizePromise = function (resolver) {
 
   promise.seqResolve = this.seqResolve;
   promise.seqReject = this.seqReject;
```
Comment (mickhansen) [inline lib/promise.js]: @janmeier do you recall why we are saving these properties? Do we need them after removing emit support?

Comment (janmeier) [inline lib/promise.js]: They are only needed for emit support, specifically 
```
  if (evt === 'success') {
    this.seqResolve.apply(this, args);
```

Comment (mickhansen) [inline lib/promise.js]: What should the constructor look like then? Can you update this PR?

Comment (janmeier) [inline lib/promise.js]: Not even sure we need to provide a SequelizePromise - modifying `Promise.prototype._then` with CLS support should be the only change neededd

**Comment by mickhansen** [comment]
@BridgeAR @janmeier still a few tests depending on `on('sql')` it seems, they need to be refactored to use `logging: ` before we can move on.

**Comment by BridgeAR** [comment]
One of the main problems here is, that there's not always an option to pass logging. In some cases  global logging is used and in others probably none at all.

As an example: 
```javascript
return self.sequelize.queryInterface.describeTable('Publics').on('sql', function(sql) { /* ... */ });
```
What do you think about providing an option that could be activated and deactivated in the test itself? That way the sql output could be tested in a generic way without changing the function without options:
```javascript
var match = false;
this.sequelize.logging = function (sql) {
    if (sql.indexOf('perfect') !== -1) {
        match = true;
    }
};
return self.sequelize.queryInterface.describeTable('Publics').then(function() {
    expect(match).to.equal(true);
    this.sequelize.logging = false;
}
```
I'm not the really a big fan of that solution but it's the first one that came to my mind.
Otherwise all functions that do not have the possibility of using logging can't be tested properly. Or is there any other possibility to handle this that I can't think of?

**Comment by janmeier** [comment]
Ideally all functions should suport the logging option. As long as the option is passed properly to `.query` everything should work from there

**Comment by BridgeAR** [comment]
In that case I'll rewrite all tests to use logging and introduce options to all functions that do not support that right now. I'm uncertain, when I'm done with that though, since I still have to finish some other tasks this week.

**Comment by janmeier** [comment]
TODO:
* Patch the global bluebird instance instead of a local one
* Mention in the docs that enabling CLS will patch the global bluebird instance

**Comment by mickhansen** [comment]
@janmeier patching the global bluebird instance is still bit of a misnomer since unless its the exact same version its local to the sequelize install.

**Comment by janmeier** [comment]
Perhaps if the user sets `Sequelize.Promise` before enabling CLS?

**Comment by mickhansen** [comment]
@janmeier Let the user inject Promise into sequelize you mean?

**Comment by janmeier** [comment]
Yep - But it feels kinda ugly and unnecessary .. Just use the same bluebird version or patch it yourself :)

**Comment by mickhansen** [comment]
Seems fair.

**Pull request merged**